FiveCalls/FiveCalls/IssuesContainerViewController.swift

```diff
@@ -65,13 +62,56 @@ class IssuesContainerViewController : UIViewController, EditLocationViewControll
     }
 
     func editLocationViewController(_ vc: EditLocationViewController, didSelectZipCode zip: String) {
+        dismiss(animated: true, completion: { //`updateWith` must be in completion block so that VC is listening for notification
+            self.updateWith(zipCode: zip)
+        })
         locationButton.setTitle(zip, for: .normal)
-        UserDefaults.standard.setValue(zip, forKey: UserDefaultsKeys.zipCode.rawValue)
-        NotificationCenter.default.post(name: .zipCodeChanged, object: nil)
-        dismiss(animated: true, completion: nil)
     }
 
-    func editLocationViewController(_ vc: EditLocationViewController, didSelectLocation location: CLLocationCoordinate2D) {
+    func editLocationViewController(_ vc: EditLocationViewController, didSelectLocation location: CLLocation) {
+        getLocationInfo(from: location) { locationInfo in
+            self.locationButton.setTitle((locationInfo["displayName"] as? String) ?? "Selected Location", for: .normal)
+            self.dismiss(animated: true, completion: {//`updateWith` must be in completion block so that VC is listening for notification
+                self.updateWith(locationInfo: locationInfo)
+            })
+        }
+    }
+
+    //Mark: private functions
 
+    private func setTitleLabel() {
+        if let zip = UserDefaults.standard.string(forKey: UserDefaultsKeys.zipCode.rawValue) {
+            locationButton.setTitle(zip, for: .normal)
+        } else if let locationInfo = UserDefaults.standard.value(forKey: UserDefaultsKeys.locationInfo.rawValue) as? [String: Any] {
+            let displayName = (locationInfo["displayName"] as? String) ?? "Selected Location"
+            self.locationButton.setTitle(displayName, for: .normal)
+        }
+    }
+
+    private func getLocationInfo(from location: CLLocation, completion: @escaping (([String: Any]) -> ())) {
+        let geocoder = CLGeocoder()
+        var locationInfo = [String: Any]()
+        locationInfo["longitude"] = location.coordinate.longitude
+        locationInfo["latitude"] = location.coordinate.latitude
+        geocoder.reverseGeocodeLocation(location, completionHandler: { (placemarks, error) in
+            let prefix = placemarks?.first?.subThoroughfare ?? ""
+            let street = placemarks?.first?.thoroughfare ?? ""
+            let streetAddress = prefix + " " + street
+            locationInfo["displayName"] = streetAddress != " " ? streetAddress : nil
+            locationInfo["zipcode"] = placemarks?.first?.postalCode ?? ""
+            completion(locationInfo)
+        })
+    }
```

Comment:
sounds great. that was my initial approach. no particular reason I didn't stick with it.